lib/src/core/verifier/mod.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use std::path::PathBuf;
+use crate::*;
+use crate::core::*;
+
+pub mod policy;
+mod sample;
+mod sgx;
+mod tdx;
+
+#[async_trait]
+pub trait TEEtraits {
+    async fn evaluate(&self, evidence: &Evidence) -> Result<AttestationResults>;
+    fn opa_policy_path(&self) -> Result<PathBuf>;
+    fn opa_reference_data_path(&self) -> Result<PathBuf>;
+}
+
+#[derive(Debug)]
+pub struct Verifiers {
+    pub tdx: tdx::Tdx,
+    pub sgx: sgx::Sgx,
+    pub sample: sample::Sample,
+}
+
+impl Verifiers {
+    pub async fn new(workdir: PathBuf) -> Result<Self> {
+        Ok(Self {
+            tdx: tdx::Tdx::new(workdir.clone()).await?,
+            sgx: sgx::Sgx::new(workdir.clone()).await?,
+            sample: sample::Sample::new(workdir.clone()).await?,
+        })
+    }
+}
+
+#[async_trait]
+impl Verifier for Verifiers {
+    async fn evaluate(&self, evidence: &Evidence) -> Result<AttestationResults> {
```

Comment:
It seems like this functionality could just go in the `Attestation` struct. Do we really need to have a separate `Verifiers` struct?

Reply:
I will add a new trait named `opa_files_path()` to export these specific TEE's OPA related interfaces unified. And I think better to keep this trait, because maybe other new trait functions will be added in future.